templates/next/components/foo/Form.tsx

```diff
@@ -0,0 +1,209 @@
+import { FunctionComponent, useState } from "react";
+import { Formik } from "formik";
+import { {{{ucf}}} } from '../../interfaces/{{{ucf}}}';
+import Link from "next/link";
+
+interface Props {
+  {{{lc}}}?: {{{ucf}}};
+}
+
+export const Form: FunctionComponent<Props> = ({ {{{lc}}} }) => {
+	const [error, setError] = useState(null);
+
+	const handleDelete = () => {
+		if (window.confirm("Are you sure you want to delete this item ?")) {
+			try {
+				fetch({ {{{lc}}}['@id'] }, { method: "DELETE" });
+			} catch (error) {
+				setError("Error when deleting the resource.");
+				console.error(error);
+			}
+		}
+	};
+
+	const defaultValues = () => {
+		if ( {{{lc}}} ) {
+			return { ...{{{lc}}} };
+		} else {
+			return {
+				isbn: "",
+				title: "",
+				description: "",
+				author: "",
+				publicationDate: "",
+				reviews: "",
+			};
+		}
+	};
+
+	return (
+		<div>
+			{{{{lc}}} ? <h1>Edit {{{lc}}}['@id']</h1> : <h1>Create</h1>}
+			<Formik
+				initialValues={defaultValues()}
+				validate={(values) => {
+					const errors = {};
+					//set your validation logic here
+				}}
+				onSubmit={(values, { setSubmitting, setStatus }) => {
+					if (!{{{lc}}}["@id"]) {
+						try {
+							fetch("/{{{name}}}", {
+								method: "post",
+								body: JSON.stringify(values),
+							});
+							setStatus({
+								isValid: true,
+								msg: "Element created",
+							});
+						} catch (error) {
+							setStatus({
+								isValid: false,
+								msg: `Error ${error}`,
+							});
+						}
+					} else {
+						try {
+							fetch({{{lc}}}["@id"], {
+								method: "put",
+								body: JSON.stringify(values),
+							});
+							setStatus({
+								isValid: true,
+								msg: "Element edited",
+							});
+						} catch (error) {
+							setStatus({
+								isValid: false,
+								msg: `Error ${error}`,
```

Comment:
```suggestion
								msg: `An unexpected error occured.`,
```

We don't need to display the technical error to the end user.